feature-utils/silly-i18n/rollup-plugin.js

```diff
@@ -0,0 +1,57 @@
+import fs from "fs";
+import path from "path";
+import util from "util";
+import glob from "glob";
+
+const DIRNAME = "locales";
+const FILEEXT = ".json";
+const PLUGIN_NAME = "silly-i18n";
+const SUFFIX = `?${PLUGIN_NAME}`;
+
+export default function (options) {
+    return {
+        name: PLUGIN_NAME,
+
+        async resolveId(source, importer) {
+            if (source == `!${PLUGIN_NAME}`) {
+                for (let dir = importer; dir != (dir = path.dirname(dir)); ) {
+                    const subdir = path.join(dir, DIRNAME);
+                    const exists = await fs.promises.stat(subdir).then(
+                        (stat) => stat.isDirectory(),
+                        () => false
+                    );
+                    if (exists) return `${subdir}${SUFFIX}`;
+                }
+                throw new Error(`"${DIRNAME}" directory not found`);
```

Comment:
We'll be getting here if none of the directory exists, I guess; gut the fact that that directory is or is not found, can't it be checked in advance?

Reply:
I don't understand what you are suggesting? How could we check that the directory exists before trying to find it?

Reply:
I don't understand why this is "outdated", but this is a different thing. I'll go to wherever it is now...

Reply:
What do you suggest to change here?

Reply:
`not found within ${whatever superdirs you're where you're looking for it}`. Point is, there's nothing callee-specific int his error message.
Also, if it's not found this error will be thrown in the first file where this happens, is that correct?

Reply:
I changed the error message to include the importer, then I don't need additional logic just to compile a list of all directories that were checked on the way, and the error message doesn't get confusingly long.

> Also, if it's not found this error will be thrown in the first file where this happens, is that correct?

Yup, the build will stop/fail as soon as this error is thrown.